R/unitConverter.R

```diff
@@ -50,45 +50,58 @@
   xTargetUnit <- xUnit %||% unique(data$xUnit)[[1]]
   yTargetUnit <- yUnit %||% unique(data$yUnit)[[1]]
 
-  # xUnit ----------------
+  # xUnit --------------------------
 
-  dataXUnit <- data %>%
-    dplyr::select(dplyr::matches("^x")) %>%
-    tidyr::nest(data = xValues) %>%
-    dplyr::rowwise() %>%
-    dplyr::mutate(xValues = list(
-      toUnit(
-        quantityOrDimension = xDimension,
-        values              = purrr::pluck(data, 1L),
-        targetUnit          = xTargetUnit,
-        sourceUnit          = xUnit
-      )
-    )) %>%
-    dplyr::ungroup() %>%
-    tidyr::unnest(cols = c(xValues)) %>%
-    dplyr::mutate(xUnit = xTargetUnit) %>%
-    dplyr::select(-data)
+  xDataList <- split(data, data$xUnit)
+  data <- dplyr::bind_rows(lapply(xDataList, .xUnitConverter, xTargetUnit))
 
-  # yUnit ----------------
+  # yUnit and error ----------------
 
-  dataYUnit <- data %>%
-    dplyr::select(-dplyr::matches("^x")) %>%
-    tidyr::nest(data = c(yValues)) %>%
-    dplyr::rowwise() %>%
-    dplyr::mutate(yValues = list(
-      toUnit(
-        quantityOrDimension = yDimension,
-        values              = purrr::pluck(data, 1L),
-        targetUnit          = yTargetUnit,
-        sourceUnit          = yUnit,
-        molWeight           = molWeight,
-        molWeightUnit       = ospUnits$`Molecular weight`$`g/mol`
-      ))) %>%
-        dplyr::ungroup() %>%
-        tidyr::unnest(cols = c(yValues)) %>%
-        dplyr::mutate(yUnit = yTargetUnit) %>%
-        dplyr::select(-data)
+  yDataList <- split(data, data$yUnit)
+  data <- dplyr::bind_rows(lapply(yDataList, .yUnitConverter, yTargetUnit))
 
-      # combine them
-      dplyr::bind_cols(dataXUnit, dataYUnit)
+  data
```

Comment:
Please add `return(data)` to all functions that return something.

Reply:
Yes, please don't review this until I mark it ready for a review.

Reply:
OK. Can I push a test for different MWs (is failing now)?